acceptance/tests/failover/timeout_failover.rb

```diff
@@ -0,0 +1,44 @@
+require 'pxp-agent/config_helper.rb'
+require 'pxp-agent/test_helper.rb'
+
+test_name 'C97964 - agent should use next broker if primary is timing out' do
+
+  blocked_broker_instance = 0
+  teardown do
+    unblock_pcp_broker(master,blocked_broker_instance)
+    broker_instance = 0 # 0 indexed
+    kill_pcp_broker(master)
+    run_pcp_broker(master, broker_instance)
+  end
+
+  step 'Ensure each agent host has pxp-agent configured with multiple uris, running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      num_brokers = 2
+      pxp_config = pxp_config_hash_using_puppet_certs(master, agent, num_brokers)
+      pxp_config['allowed-keepalive-timeouts'] = 0
+      create_remote_file(agent, pxp_agent_config_file(agent), pxp_config.to_json.to_s)
+      on(agent, "rm -rf #{logfile(agent)}")
+      on agent, puppet('resource service pxp-agent ensure=running')
+      show_pcp_logs_on_failure do
+        assert(is_associated?(master, "pcp://#{agent}/agent"),
+               "Agent identity pcp://#{agent}/agent for agent host #{agent} does not appear in pcp-broker's client inventory")
+      end
+    end
+  end
+
+  step 'Block primary broker, start replica' do
+    block_pcp_broker(master,blocked_broker_instance)
+    broker_instance = 1 # 0 indexed
+    run_pcp_broker(master, broker_instance)
+  end
+
+  step 'On each agent, test that a new association has occurred' do
```

Comment:
Similar to earlier comment; the step string, assertion statement, and assertion string should all specify that we are asserting association on the failover broker; even though I can see that the helper methods in this case are going to use the failover broker.

At this point it would be valuable to assert that we are not associated with the primary broker any longer. Failover is poor if we did connect to the replica but didn't cleanly disconnect from the primary. That might need to be a separate step, as it would need to be unblocked so it can respond to inventory_requests (or, unless the iptables rules are made more specific so that pxp-agent is cut off from the broker, but ruby-pcp-client can still connect and query the inventory while pxp-agent is cut off)